lib/shrine/attachment.rb

```diff
@@ -34,30 +34,28 @@ def initialize(name, **options)
       # Defines attachment methods for the specified attachment name. These
       # methods will be added to any model that includes this module.
       def define_attachment_methods!
-        attachment    = self
-        attacher_ivar = :"@#{@name}_attacher"
```

Comment:
me too. this is @janko-m's suggestion: https://github.com/shrinerb/shrine/issues/322#issuecomment-439824950

Reply:
Maybe he'll reconsider.

Reply:
@hmistry If we have `attacher_ivar` local variable holding `:@#{name}_attacher`, which is used in the first method, then to be consistent we should also have `attacher_smth` local variable to hold `:#{name}_attacher` which is used in other methods, no? That's one of the reasons I don't want this change, because I want to have only one local variable.

Another reason is that in general I don't like to extract things outside of `define_method` blocks. I prefer to contain as much of the implementation inside the method, hence only the `name` locale variable.

Reply:
@janko-m Yes to answer your question. The `attacher_ivar` will be within the define_method block and only `named_attacher` will be external. 
I brought it up because we're changing from current and it felt like undoing a small optimization but your reasoning of minimizing things outside of blocks is a good point.